logging_cache.go

```diff
@@ -113,3 +119,30 @@ func (c *loggingCacheImpl) Remove(id string) {
 
 	delete(c.cache, id)
 }
+
+func (c *loggingCacheImpl) CloseAndRemove(id string) error {
+	c.mu.Lock()
+	defer c.mu.Unlock()
+
+	var err error
+	logger, ok := c.cache[id]
+	if ok {
+		err = logger.Close()
```

Comment:
Can't we just call `c.Remove(id)` on L129 and then close the logger in the the ok block? It would get out of passing the error out of the ok block.

Reply:
That would cause a deadlock, I could just return the error from the ok block and return nil outside, but I thought it was a bit more concise to have one return statement. I guess it is just a stylistic choice at this point.